vMenu/menus/VehicleOptions.cs

```diff
@@ -1702,7 +1704,25 @@ async void HandleItemSelect(Menu menu, MenuItem menuItem, int itemIndex)
                 if (vehicleExtras.TryGetValue(item, out var extra))
                 {
                     var veh = GetVehicle();
-                    veh.ToggleExtra(extra, _checked);
+                    if (veh != null && veh.Exists())
+                    {
+                        // If vmenu_prevent_extras_when_damaged is enabled, check vehicle engine and body health
+                        if (GetSettingsBool(Setting.vmenu_prevent_extras_when_damaged) && 
+                        (API.GetVehicleBodyHealth(veh.Handle) < GetSettingsInt(Setting.vmenu_prevent_extras_engine_damage) || 
+                        API.GetVehicleEngineHealth(veh.Handle) < GetSettingsInt(Setting.vmenu_prevent_extras_body_damage)))
+                        {
+                            if (GetSettingsBool(Setting.vmenu_prevent_extras_notification_enabled))
+                            {
+                                Screen.ShowNotification(GetSettingsString(Setting.vmenu_prevent_extras_notification_text));
```

Comment:
Following on from my prior comment, please make this a hard-coded string. Further, please use the existing `Notify` class instead of using `Screen`. Additionally, as vMenu doesn't provide a "mechanic", it would make more sense to simply inform the player to repair the vehicle.

For example:

```c#
Notify.Alert("Vehicle is too damaged to change extra, repair it first!", true, false);
```